src/main/java/bio/terra/service/duos/DuosDao.java

```diff
@@ -81,6 +80,16 @@ public List<DuosFirecloudGroupModel> retrieveFirecloudGroups() {
     return jdbcTemplate.query(DUOS_FIRECLOUD_GROUP_QUERY, DUOS_FIRECLOUD_GROUP_MAPPER);
   }
 
+  @Transactional(
+      propagation = Propagation.REQUIRED,
+      isolation = Isolation.SERIALIZABLE,
+      readOnly = true)
+  public List<DuosFirecloudGroupModel> retrieveFirecloudGroups(List<UUID> ids) {
+    String sql = DUOS_FIRECLOUD_GROUP_QUERY + " WHERE id IN (:ids)";
```

Comment:
Does it make sense to add a check here to make sure that ids isn't empty (and just return an empty list if it is?)

Reply:
Not a bad idea, I can add this… no point in hitting the DB if not necessary.